src/lib/work_selector/inputs.ml

```diff
@@ -29,12 +29,13 @@ module Test_inputs = struct
 
   module Snark_pool = struct
     module T = struct
-      type t = Ledger_proof.t list [@@deriving bin_io, hash, compare, sexp]
+      type t = Transaction_snark.Statement.Stable.Latest.t list
```

Comment:
`Ledger_proof.t` is only equivalent to `Transaction_snark.Statement.t` when the proof level is set to full, right? Did you have to make another change to make this work?

Reply:
A `Transaction_snark.Statement.t` is the parameters to the proposition proved by a transaction snark i.e. the source ledger hash + the target ledger hash + some misc data. In these test inputs, we use statements instead of proofs, to avoid doing the actual proving.

Reply:
I'm just confused how before this type was different based on the build profile you select where as now the type is always the same (take a look at `ledger_proof.ml` to see what I mean; depending on the value of `proof_level`, it either includes `Prod` or `Debug`, which have different types). If it compiles, it compiles, just wanted to make sure there is not logical reason why they had to have different types based on the proof level at this context.

Reply:
Oh, I see what you mean. Name shadowing is confusing us here, that `Ledger_proof` refers to the mocked module above. I'll investigate some more.

Reply:
The old tests are completely nonsensical. A `Snark_pool.t` is a hashtable mapping `Ledger_proof.t list`s to `Transaction_snark_work.t`s, but `Ledger_proof.t` and `Transaction_snark_work.t` are both mocked to `Currency.Fee.t`. So it maps `Currency.Fee.t list`s to `Fee.t`s. The new tests test the right thing, mapping transaction snark statements to fees.